adding-data/index.md

```diff
@@ -0,0 +1,227 @@
+
+## Examples
+
+### Kyve-Solana Integration
+
+As an example, let's walk through the Solana integration. The `KyveSolana` class is as below. 
+
+```ts
+
+class KyveSolana extends KYVE {
+  public async getDataItem(key: number): Promise<{ key: number; value: any }> {
+    let block;
+
+    try {
+      block = await fetchBlock(
+        this.pool.config.rpc,
+        key,
+        await this.getSignature()
+      );
+    } catch (err) {
+      if (wasSlotSkipped(err, key)) return { key, value: null };
+
+      this.logger.warn(
+        `⚠️  EXTERNAL ERROR: Failed to fetch block ${key}. Retrying ...`
+      );
+
+      throw err;
+    }
+
+    return { key, value: block };
+  }
+
+  private async getSignature(): Promise<Signature> {
+   // ...
+  }
+}
+```
+
+Let's take a closer look at the `fetchBlock` helper method, which in this case lives in its own `utilies.ts` file.
+
+```ts
+import { BlockResponse, Connection } from '@solana/web3.js';
+
+// NOTE: The response type isn't correct because of our patch.
+export async function fetchBlock(
+  endpoint: string,
+  height: number,
+  signature: Signature
+): Promise<BlockResponse> {
+  const provider = initialiseSolanaRPC(endpoint, signature);
+
+  return (await provider.getBlock(height))!;
+}
+
+function initialiseSolanaRPC(
+  endpoint: string,
+  signature: Signature
+): Connection {
+  return new Connection(endpoint, {
+    httpHeaders: {
+      'Content-Type': 'application/json',
+      Signature: signature.signature,
+      'Public-Key': signature.pubKey,
+      'Pool-ID': signature.poolId,
+      Timestamp: signature.timestamp,
+    },
+  });
+}
+```
+
+The `fetchBlock` method first initializes a connection to the Solana API and then calls the `getBlock()` method that is provided in the Solana SDK.
+
+### Kyve-Near Integration
+
+The `fetchBlock` helper method for the Kyve-Near integration fetches data from the source chain (Near) directly from Near's RPC endpoints.
+
+```ts
+export async function fetchBlock(
+  endpoint: string,
+  height: number,
+  signature: Signature
+): Promise<any> {
+  const res = await call<any>(
+    `${endpoint}/cosmos/base/tendermint/v1beta1/blocks/${height}`,
+    signature
+  );
+
+  const txs = await fetchTransactions(
+    endpoint,
+    res.block.data.txs.map(parseEncodedTx),
+    signature
+  );
+
+  return {
+    ...res.block,
+    data: {
+      ...res.block.data,
+      txs,
+    },
+  };
+}
+```
+
+## Custom Logic
+
+In addition to the `getDataItem` method which must be implemented, you are free to implement custom methods in your class. 
+
+### Custom Signatures
+
+For example, for several of the existing integrations, we have implemented custom signatures.
+
+These signatures, which were implemented to prevent spamming of private Infura endpoints, are calculated from the address (conveniently exposed in the class through your wallet) and the message. They can be used for signature verification.
```

Comment:
```suggestion
These signatures, which were implemented to prevent spamming of private endpoints, are calculated from the address (conveniently exposed in the class through your wallet) and the message. They can be used for signature verification.
```